src/components/forms/RegisterForm.tsx

```diff
@@ -44,39 +39,63 @@ const RegisterForm = (): JSX.Element => {
     }
   }
 
-  const onFormSubmit = (e: React.FormEvent): void => e.preventDefault()
-
   return (
-    <Form onSubmit={onFormSubmit}>
+    <Form onSubmit={handleSubmit(handleRegister)}>
       <Form.Group className='mb-3'>
         <Form.Label>{t('form.userName')}</Form.Label>
         <Form.Control
-          required
+          {...register(
+            'username',
+            {
+              required: { value: true, message: t('error.requiredField') },
+              minLength: { value: minLengthUsername, message: t('error.minLengthError', { count: minLengthUsername }) }
+            })
+          }
           type='text'
-          onChange={onUsernameChange}
         />
+        {(errors.username != null) &&
+          <span className='mt-4 text-danger'>
+            {errors.username.message}
+          </span>}
       </Form.Group>
 
       <Form.Group className='mb-3'>
         <Form.Label>{t('form.emailAddress')}</Form.Label>
         <Form.Control
-          required
+          {...register(
+            'email',
+            {
+              required: { value: true, message: t('error.requiredField') }
+            })
+          }
           type='email'
-          onChange={onEmailChange}
         />
+        {(errors.email != null) &&
+          <span className='mt-4 text-danger'>
+            {errors.email.message}
+          </span>}
```

Comment:
The way this is handled can be refactored to its own component so the code/logic is the same for username, email and password. The component takes the errors.field in parameter

Reply:
I see if you prefer

Reply:
done